internal/envoy/route.go

```diff
@@ -294,6 +298,43 @@ func RouteConfiguration(name string, virtualhosts ...*envoy_api_v2_route.Virtual
 	}
 }
 
+// CorsPolicy returns a *v2.CorsCorsPolicy
+func CorsPolicy(cp *dag.CorsPolicy) *envoy_api_v2_route.CorsPolicy {
+	if cp == nil {
+		return nil
+	}
+	rcp := &envoy_api_v2_route.CorsPolicy{
+		EnabledSpecifier: &envoy_api_v2_route.CorsPolicy_FilterEnabled{
+			FilterEnabled: &envoy_api_v2_core.RuntimeFractionalPercent{
+				DefaultValue: &envoy_type.FractionalPercent{
+					Numerator:   100,
+					Denominator: envoy_type.FractionalPercent_HUNDRED,
+				},
+			},
+		},
+		AllowCredentials: &wrappers.BoolValue{
+			Value: cp.AllowCredentials,
+		},
+		AllowHeaders:  strings.Join(cp.AllowHeaders, ","),
+		AllowMethods:  strings.Join(cp.AllowMethods, ","),
+		ExposeHeaders: strings.Join(cp.ExposeHeaders, ","),
+	}
+
+	if !cp.MaxAge.UseDefault() {
+		rcp.MaxAge = fmt.Sprintf("%.0f", cp.MaxAge.Duration().Seconds())
+	}
```

Comment:
If you want to support disabling MaxAge (see other comment), then you'd also want to check for `cp.MaxAge.IsDisabled()` here, and if true, pass a value of `-1` to Envoy.

Reply:
Now I'm doing that but for the `0` value.